Publish YDB CLI 2.18.0 (#13104)
ydb-platform/ydb: ydb/docs/en/core/changelog-cli.md

# {{ ydb-short-name }} CLI changelog

## Version 2.18.0 {#2-18-0}

Released on December 24, 2024. To update to version **2.18.0**, select the [Downloads](downloads/index.md#ydb-cli) section.

> * Query plan and statistics improvements:
>   * Expression and attributes added to various operator properties (i.e. GroupBy)
>   * Per operator statistics (Rows)
>   * Statistics from Column Shards (Rows and Bytes)

Reviewer: ```suggestion
* Query plan and statistics provide additional information:
  * Expression and attributes added to various operator properties (e.g., `GroupBy`)
  * Per-operator statistics (Rows)
  * Statistics for [column-oriented tables](./concepts/datamodel/table.md#column-oriented-tables) (Rows and Bytes)
```

>   * Statistics from Column Shards (Rows and Bytes)

Reviewer: Clarify what those Rows and Bytes mean. Probably rows processed for operators and rows/bytes read from column shards, but I had to guess that.

> * Fixed a bug where `ydb workload * run` command could crash in `--dry-run` mode.

Reviewer: ```suggestion
* Fixed a bug where the `ydb workload * run` command could crash in `--dry-run` mode.
```

> * Added support for views in local backups: `ydb tools dump` and `ydb tools restore`. Views are backed up as `CREATE VIEW` queries saved in the `create_view.sql` files, which can be executed to recreate the original views.

Reviewer: Was there a PR adding this to the actual documentation for these commands?

> * Added support for views in local backups: `ydb tools dump` and `ydb tools restore`. Views are backed up as `CREATE VIEW` queries saved in the `create_view.sql` files, which can be executed to recreate the original views.
> * Replaced option `--query-settings` by `--query-prefix` one in `ydb workload <workload> run`.

Reviewer: ```suggestion
* Added support for views in local backups: `ydb tools dump` and `ydb tools restore`. Views are backed up as `CREATE VIEW` queries saved in the `create_view.sql` files, which can be executed to recreate the original views.
* Replaced the `--query-settings` option with `--query-prefix` in `ydb workload <workload> run`.
```

> * Replaced option `--query-settings` by `--query-prefix` one in `ydb workload <workload> run`.

Reviewer: This needs to be marked as a backward-incompatible change if it doesn't support the old name anymore.

> * Added new options to `ydb workload topic`: --tx-commit-interval and --tx-commit-messages, allowing you to specify commit interval either in milliseconds or in number of messages written.
> Also now you can load test YDB topics, using wide transactions that span over all partitions in the topic. This works both in write and in end-to-end workload scenarios.

Reviewer: ```suggestion
* Added new options to `ydb workload topic`: `--tx-commit-interval` and `--tx-commit-messages`, allowing you to specify the commit interval in milliseconds or by the number of messages written.
* Added support for load testing [{{ ydb-short-name }} topics](./concepts/topic.md) using wide transactions that span all partitions in the topic. This works in both write and end-to-end workload scenarios.
```

> Also now you can load test YDB topics, using wide transactions that span over all partitions in the topic. This works both in write and in end-to-end workload scenarios.

Reviewer: Needs a link to documentation on how to run this workload.

> * `ydb import file csv` command now saves import progress. Relaunching import command will continue from the line it was interrupted on
> * Use QueryService by default (`--executer generic`) in `ydb workload kv` and `ydb workload stock` commands

Reviewer: ```suggestion
* The `ydb import file csv` command now saves import progress. Relaunching the import command will continue from the row where it was interrupted.
* In the `ydb workload kv` and `ydb workload stock` commands, the default value of the `--executer` option has been changed to `generic`, which makes them no longer rely on the legacy query execution infrastructure.
```

> * Use parquet format instead of CSV to fill tables in `ydb workload` benchmarks
> * Made `--consumer` flag in `ydb topic read` command optional. Now if this flag is not specified, reading is performed in no-consumer mode. In this mode partition IDs should be specified with `--partition-ids` option.

Reviewer: ```suggestion
* Replaced the CSV format with Parquet for filling tables in `ydb workload` benchmarks.
* Made the `--consumer` flag in the `ydb topic read` command optional. If this flag is not specified, reading is performed in no-consumer mode. In this mode, partition IDs must be specified with the `--partition-ids` option.
```

> * Fixed a bug in `ydb import file csv` where multiple columns with escaped quotes in the same row were parsed incorrectly
> * Truncate query results output in benchmarks
> * Added `ydb admin cluster bootstrap` command to bootstrap automatically configured cluster

Reviewer: ```suggestion
* Fixed a bug in `ydb import file csv` where multiple columns with escaped quotes in the same row were parsed incorrectly.
* Truncated query results output in benchmarks.
* Added the `ydb admin cluster bootstrap` command to bootstrap an automatically configured cluster.
```

> * Added `ydb admin cluster bootstrap` command to bootstrap automatically configured cluster

Reviewer: Afaik, this command would work only with a server built from main or 25.1+

Reviewer 2: Саша Рутковский говорит, что должно уже сейчас работать.
